src/Microsoft.Health.Fhir.Shared.Api/Controllers/ImportController.cs

```diff
@@ -79,9 +93,80 @@ public class ImportController : Controller
             _urlResolver = urlResolver;
             _features = features.Value;
             _mediator = mediator;
+            _resourceWrapperFactory = EnsureArg.IsNotNull(resourceWrapperFactory, nameof(resourceWrapperFactory));
             _logger = logger;
         }
 
+        [HttpPost]
+        [Route(KnownRoutes.ImportBundle)]
+        [AuditEventType(AuditEventSubType.ImportBundle)]
+        public async Task<IActionResult> ImportBundle()
+        {
+            var sw = Stopwatch.StartNew();
+            var startDate = DateTime.UtcNow;
+            var resources = new List<ImportResource>();
+            var keys = new HashSet<ResourceKey>();
+            var fhirParser = new FhirJsonParser();
+            var importParser = new ImportResourceParser(fhirParser, _resourceWrapperFactory);
+            var index = 0L;
+            using var reader = new StreamReader(Request.Body, Encoding.UTF8);
+            try
+            {
+                if (Request.ContentType != null && Request.ContentType.Contains("application/fhir+ndjson", StringComparison.OrdinalIgnoreCase))
+                {
+                    var line = await reader.ReadLineAsync();
```

Comment:
What is the stream is really large, should we have some validation or limits?

Reply:
I will make the change.